### DISCUSSION TITLE
Orchard adds spaces to my content part names - why, and how to disable?

### DISCUSSION THREAD

**Discussion by andrerav** [OrchardCMS/OrchardCore#11600] [Q&A]
Hi,
When I create a content part with the name "SomeKindOfContentComponent", the technical name is split up at the uppercased characters, like so:

![image](https://user-images.githubusercontent.com/3635018/165540128-e3a8e3d8-7891-45aa-9951-fd7b9e62fe50.png)

Why is this, and how can I permanently disable it?

**Comment by hishamco** [comment]
What it shown is a display name, but I'm surprsing that the display name in your pic don't have space?!!

**Reply by andrerav** [reply]
Why would it add spaces to the display name? How can I disable this? In my native language this can create some extremely confusing experiences for the user.

**Reply by hishamco** [reply]
No, I mean the display name could contains space, that's why I'm suprised when I saw the display name without spaces and appears with space

One question are you working on the code base or a specific verion?

**Reply by andrerav** [reply]
> No, I mean the display name could contains space, that's why I'm suprised when I saw the display name without spaces and appears with space

Sorry, I don't understand what you are trying to say here.

> One question are you woring on the code base or a specific verion?

Sorry, I don't understand this question either. Could you be more specific?

**Reply by hishamco** [reply]
There's a typo ;)  which OC version you are using? Or you are working on the source code?

**Reply by andrerav** [reply]
I did the following steps to create my Orchard Core instance:

```powershell
dotnet new -i "OrchardCore.ProjectTemplates::*"
dotnet new occms --name "Lk.hCMS"
dotnet new sln --name "Lk.hCMS"
dotnet sln add .\Lk.hCMS\Lk.hCMS.csproj
(Get-Content -path .\Lk.hCMS\Lk.hCMS.csproj -Raw) -replace '-preview-16386',''| Set-Content -Path .\Lk.hCMS\Lk.hCMS.csproj
cd .\Lk.hCMS\
dotnet run .
```

**Reply by andrerav** [reply]
So that means I am using v1.3.0 which should be the latest stable.

**Comment by Skrypt** [comment]
The Technical name is generated from the Display Name. It is using Camel casing intentionally by convention. When you are refering to your content type from code you can't use the display name with spaces. This is why we need a technical value that can be used as a variable from code. This can't be disabled because it is a technical value that we use from code to reference to a specific Content Type. I'm not sure what you mean by saying that this can cause confusion since for users since this technical value can't be changed from the admin UI and should be only used by developpers that need to make Queries or Create Content Types from code.

Here, when you changed the display name to match the technical name it doesn't make any difference. Your technical name could be "foo" while the display name would be "bar". This is up to you to change the technical value from database migrations if it is needed to cause less confusion from people.

**Reply by andrerav** [reply]
I think you may have misunderstood me.
> Here, when you changed the display name

I did not change anything. I just wrote "SomeKindOfContentComponent" as the content part name. I did not add the spaces. Orchard Core did. There is no code involved here. I have not written a single line of code. I just used the UI to create the content part. Orchard Core adds spaces when displaying it, as seen above. I want to disable this behaviour.

**Comment by hishamco** [comment]
@Skrypt the issue is he providing a display name similar to the technical one, surprisngly it shows with spaces, which is confuses me ;)

This is the first time I saw such behavior, I will try to reproduce in the code base ..

**Comment by Skrypt** [comment]
Here are the repro steps. It happens with Content Parts. I thought it was with Content Types. I did not read carefully. Sorry.

https://user-images.githubusercontent.com/3228637/165676824-b8ad6501-a9dd-4969-a2f0-fa42fa7db727.mp4

Please open an issue.

**Reply by hishamco** [reply]
Issue created #11610

**Reply by hishamco** [reply]
I may have a look to this issue, BTW @Skrypt what are using for create such mp4 clips?

**Reply by Skrypt** [reply]
ScreenToGif

**Reply by hishamco** [reply]
I already have it. I though it produces a GIF, let me have a try

Thanks

**Reply by Skrypt** [reply]
You need to enable the FFMpeg extension and you can then export as an mpg file.

**Reply by hishamco** [reply]
Finally it works, thanks
